pkg/gui/password.go

```diff
@@ -36,10 +44,10 @@ func PasswordDialog(nickname string, app *fyne.App) chan string {
 		Items: items,
 		OnSubmit: func() {
 			window.Hide()
-			passwordText <- password.Text
+			passwordEntry <- PasswordEntry{password.Text, nil}
 		},
 		OnCancel: func() {
-			passwordText <- ""
+			passwordEntry <- PasswordEntry{password.Text, errors.New("canceled by user")}
```

Comment:
A couple of things here:
- when you use composite litteral, I personnaly find it easier to use labels. That way I don't have to get the struct in mind. More information on composite litterals [here](https://go.dev/doc/effective_go#composite_literals).
- using password.Text instead of an empty string makes me wonder what is the default value of password.Text. And i don't see it in the screen.

Mu suggestion is:
```suggestion
			passwordEntry <- PasswordEntry{"", errors.New("password entry cancelled by the user")}
```